packages/playwright-core/src/client/page.ts

```diff
@@ -200,6 +202,17 @@ export class Page extends ChannelOwner<channels.PageChannel> implements api.Page
     await this._browserContext._onRoute(route);
   }
 
+  private async _onWebSocketRoute(webSocketRoute: WebSocketRoute) {
+    const routeHandlers = this._webSocketRoutes.slice();
+    for (const routeHandler of routeHandlers) {
+      if (!routeHandler.matches(webSocketRoute.url()))
+        continue;
+      await routeHandler.handle(webSocketRoute);
```

Comment:
maybe we should throw here:
- after the handler was executed
- no connect() was called
- the user called routeReceive/routeSend
That might help a few users.